specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,200 @@
+                        Microsoft::WRL::ComPtr<ICoreWebView2WebResourceResponse> response;
+                        CHECK_FAILURE(environment->CreateWebResourceResponse(
+                            response_stream.Get(), 200, L"OK", L"Content-Type: image/jpeg",
+                            &response));
+                        args->put_Response(response.Get());
+                    }
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_webResourceRequestedToken);
+```
+
+# API Details
+## IDL
+```c#
+/// Specifies the source of `WebResourceRequested` event.
+[v1_enum]
+typedef enum COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE {
+  /// Indicates that web resource is requested from main page including dedicated workers and iframes.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_DOCUMENTS = 1,
+
+  /// Indicates that web resource is requested from shared worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SHARED_WORKER = 2,
+
+  /// Indicates that web resource is requested from service worker.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_SERVICE_WORKER = 4,
+
+  /// Indicates that web resource is requested from any supported source.
+  COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE_ALL = 0XFFFFFFFF
+} COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_WEB_RESOURCE_REQUEST_SOURCE);")
+
+[uuid(1cc6a402-3724-4e4a-9099-8cf60f2f93a1), object, pointer_default(unique)]
+interface ICoreWebView2_16: ICoreWebView2_15 {
+  /// A web resource request with a resource context that matches this
+  /// filter's resource context, an URI that matches this filter's URI
+  /// wildcard string for corresponding request source will be raised via
+  /// the `WebResourceRequested` event. To receive all raised events filters have
+  /// to be added before main page navigation.
+  ///
+  /// The `uri` parameter value is a wildcard string matched against the URI
+  /// of the web resource request. This is a glob style
+  /// wildcard string in which a `*` matches zero or more characters and a `?`
+  /// matches exactly one character.
+  /// These wildcard characters can be escaped using a backslash just before
+  /// the wildcard character in order to represent the literal `*` or `?`.
+  ///
+  /// The matching occurs over the URI as a whole string and not limiting
+  /// wildcard matches to particular parts of the URI.
+  /// The wildcard filter is compared to the URI after the URI has been
+  /// normalized, any URI fragment has been removed, and non-ASCII hostnames
+  /// have been converted to punycode.
+  ///
+  /// Specifying a `nullptr` for the uri is equivalent to an empty string which
+  /// matches no URIs.
+  ///
+  /// For more information about resource context filters, navigate to
+  /// [COREWEBVIEW2_WEB_RESOURCE_CONTEXT].
+  ///
+  /// Because service worker runs separately from any one HTML document its WebResourceRequested
```

Comment:
also shared workers, correct? 
```suggestion
  /// Because service workers and shared workers run separately from any one HTML document, its WebResourceRequested
```

Reply:
Ok, you are right - it is how shared workers behave now also - we receive events in both webview2s if they are subscribed even if it is initiated only from the one of them. But I'm wondering if I need to fix this and have shared worker event send to only webview which started it - similar to how it works for main page and iframes? Or it is correct and expected? CC: @LiangTheDev

Reply:
Shared workers are shared amongst different browsing contexts and so can be used by different CoreWebView2s similarly to service workers so makes sense to me that they raise events on multiple CoreWebView2s (or at least that the behavior matches service workers)